cli/src/pcluster/models/cluster_config.py

```diff
@@ -1350,29 +1352,105 @@ def _validate(self):
                 )
 
 
-class ClusterBucket:
+# ------------ Attributes class used in cluster resources ----------- #
+
+
+class ClusterBucket(S3Bucket):
     """Represent the cluster s3 bucket configuration."""
 
     def __init__(
         self,
-        name: str,
-        artifact_directory: str,
-        remove_on_deletion: bool,
+        cluster_name: str,
+        stack_name: str,
+        name: str = None,
+        artifact_remove_on_deletion: bool = True,
     ):
-        super().__init__()
-        self.name = name
-        self.artifact_directory = artifact_directory
-        self.remove_on_deletion = remove_on_deletion
+        super().__init__(artifact_remove_on_deletion, name)
+        self.cluster_name = cluster_name
+        self.stack_name = stack_name
+        self.root_cluster_directory = "clusters"
+        self.source_config_name = "cluster-config-original.yaml"
+        self.config_name = "cluster-config.yaml"
+        self.template_name = "aws-parallelcluster.cfn.yaml"
+        self.artifacts_name = "artifacts.zip"
+        self.instance_types_data_name = "instance-types-data.json"
+        self.cluster_directory = None
+        self.full_cluster_dir = None
```

Comment:
are these all supposed to be "public" properties the caller can access? it might be good to encapsulate the structure of the bucket in this class and for example make dir and file names some local constants that can also be defined at the cluster level with a leading underscore to mark them as "private"

Reply:
I meant at the class level not cluster, sorry :)